latch_cli/services/init/init.py

```diff
@@ -1,142 +1,172 @@
 """Service to initialize boilerplate."""
 
+import json
 import os
 import re
 import shutil
 from enum import Enum, auto
+from importlib.metadata import version
 from pathlib import Path
 from textwrap import dedent
-from typing import Optional
+from typing import Callable, Optional
+
+import click
+
+from latch_cli.tui import select_tui
 
 
 class _Templates(Enum):
-    default = auto()
+    empty = auto()
+    subprocess = auto()
     r = auto()
     conda = auto()
 
 
-def init(pkg_name: Path, template: Optional[str] = None):
+def init(pkg_name: str, expose_dockerfile: bool = True) -> bool:
     """Creates boilerplate workflow files in the user's working directory.
 
     Args:
         pkg_name: A identifier for the workflow - will name the boilerplate
             directory as well as functions within the constructed package.
-        template: Which (if any) template to use when generating the files.
-            Valid choices are 'default', 'r', or 'conda'. If `template` is None,
-            the package will be generated using the 'default' template.
+        expose_dockerfile: Whether to expose a Dockerfile in the workflow.
+            If true, the Dockerfile will be created at init time and can be
+            modified. Otherwise, the Dockerfile will be created at registration
+            time and the user will not be able to modify it.
 
     Example:
 
         >>> init("test-workflow")
             # The resulting file structure will look like
             #   test-workflow
             #   ├── Dockerfile
-            #   ├── reference
-            #   │   ├── wuhan.1.bt2
-            #   │   ├── wuhan.2.bt2
-            #   │   ├── wuhan.3.bt2
-            #   │   ├── wuhan.4.bt2
-            #   │   ├── wuhan.fasta
-            #   │   ├── wuhan.rev.1.bt2
-            #   │   └── wuhan.rev.2.bt2
             #   ├── version
             #   └── wf
             #       └── __init__.py
 
     """
-    # click doesn't support enums for options hence '.name' madness
 
-    # Workflow name must not contain capitals or end in a hyphen or underscore. If it does, we should throw an error
-    # Check for capitals
-    if any(char.isupper() for char in str(pkg_name)):
+    using_dir_name = False
+    if pkg_name == ".":
+        curdir = Path(os.getcwd())
+        pkg_name = curdir.name
+        using_dir_name = True
```

Comment:
We should support all relative paths. Wanted to support this right now because it is very annoying.